clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -976,6 +976,40 @@ void CodeGenModule::Release() {
       Context.getTypeSizeInChars(Context.getWideCharType()).getQuantity();
   getModule().addModuleFlag(llvm::Module::Error, "wchar_size", WCharWidth);
 
+  if (getTriple().isOSzOS()) {
+    getModule().addModuleFlag(llvm::Module::Warning, "Product Major Version",
+                              uint32_t(CLANG_VERSION_MAJOR));
+    getModule().addModuleFlag(llvm::Module::Warning, "Product Minor Version",
+                              uint32_t(CLANG_VERSION_MINOR));
+    getModule().addModuleFlag(llvm::Module::Warning, "Product Patchlevel",
+                              uint32_t(CLANG_VERSION_PATCHLEVEL));
```

Comment:
I think I commented on that previously:  these flags are set only on z/OS, but have rather generic names.   If they're z/OS specific, I think they should have names in the "zos_..." name space like the others.

Reply:
Done. All your remaining comments have also been addressed.